crates/distribution-types/src/annotation.rs

```diff
@@ -0,0 +1,52 @@
+use serde::{Deserialize, Deserializer, Serialize};
+use std::path::PathBuf;
+use uv_fs::Simplified;
+
+/// Source of a dependency, e.g., a `-r requirements.txt` file.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize)]
+pub enum SourceAnnotation {
```

Comment:
I renamed this because we have some other "source" concepts and I wanted to disambiguate it.